### PR TITLE
More spawn/sleep removals

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
Also I added a helper proc for making things dance to reduce code duplication.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/guardian/types/lightning.dm [hunk 1]
```diff
@@ -107,12 +107,10 @@
...
 							H.electrocution_animation(20)
-						C.jitteriness += 1000
-						C.do_jitter_animation(jitteriness)
+						C.Jitter(1000)
+						C.do_jitter_animation(1000)
 						C.stuttering += 1
-						spawn(20)
-							if(C)
-								C.jitteriness = max(C.jitteriness - 990, 10)
+						addtimer(CALLBACK(C, /mob.proc/Jitter, 10), 2 SECONDS, TIMER_UNIQUE|TIMER_OVERRIDE)
```
Comment (nemvar) [inline code/modules/mob/living/simple_animal/guardian/types/lightning.dm]: This entire section is copypasta from electrocute_act. There should be a specific proc to end this in carbon_defense. This works fine enough as is but this entire section triggers me immensely.

File: code/game/machinery/dance_machine.dm [hunk 2]
```diff
@@ -312,15 +312,15 @@
...
-	for(var/i = 1, i < 10, i++)
-		for(var/d in list(NORTH,SOUTH,EAST,WEST,EAST,SOUTH,NORTH,SOUTH,EAST,WEST,EAST,SOUTH))
-			M.setDir(d)
-			if(i == WEST)
-				M.emote("flip")
-			sleep(1)
+/obj/machinery/jukebox/disco/proc/dance2(mob/living/M)
+	for(var/i in 0 to 9)
+		dance_rotate(M, CALLBACK(M, /mob.proc/dance_flip))
 		sleep(20)
```
Comment (Fikou) [inline code/game/machinery/dance_machine.dm]: is this supposed to still be here

Comment (spookydonut) [reply code/game/machinery/dance_machine.dm]: yeah before it did 10 lots of the dancing and now it still does (i think)

File: code/modules/mob/living/simple_animal/guardian/types/lightning.dm (hunk 1 above)
Comment (spookydonut) [reply code/modules/mob/living/simple_animal/guardian/types/lightning.dm]: is it? i'll maybe see if i can combine them

**Pull request merged**